src/main/java/sjchat/users/UserServiceServer.java

```diff
@@ -0,0 +1,89 @@
+package sjchat.users;
+
+import java.io.IOException;
+import java.util.Random;
+
+import io.grpc.Server;
+import io.grpc.ServerBuilder;
+import io.grpc.stub.StreamObserver;
+
+public class UserServiceServer {
+  private Server server;
+  private int port = 50051;
+
+  public static void main(String[] args) throws Exception {
+    UserServiceServer userServiceServer = new UserServiceServer();
+    userServiceServer.start();
+    userServiceServer.blockUntilShutdown();
+  }
+
+  private void start() throws IOException {
+    ;
+    server = ServerBuilder.forPort(port).addService(new UserService()).build().start();
+    System.out.println("Server started, listening on " + port);
+    Runtime.getRuntime().addShutdownHook(new Thread() {
+      @Override
+      public void run() {
+        System.err.println("*** shutting down gRPC server since JVM is shutting down");
+        UserServiceServer.this.stop();
+        System.err.println("*** server shut down");
+      }
+    });
+  }
+
+  private void stop() {
+    if (server != null) {
+      server.shutdown();
+    }
+  }
+
+  /**
+   * Await termination on the main thread since the grpc library uses daemon threads.
+   */
+  private void blockUntilShutdown() throws InterruptedException {
+    if (server != null) {
+      server.awaitTermination();
+    }
+  }
+
+  static class UserService extends UserServiceGrpc.UserServiceImplBase {
+
+    @Override
+    public void createUser(UserDataRequest req, StreamObserver<UserResponse> responseObserver) {
```

Comment:
This seems like it belongs in some kind of `MockUserService` rather than the actual implementation?

Same for the remaining calls.